src/libsync/theme.cpp

```diff
@@ -889,4 +889,16 @@ QColor Theme::errorBoxBorderColor() const
     return QColor{"black"};
 }
 
+QPalette Theme::systemPalette()
+{
+    if(!_guiAppInstance) {
+        QGuiApplication *ptr = qobject_cast<QGuiApplication *>(QGuiApplication::instance());
+        if(ptr) {
+            _guiAppInstance.reset(ptr);
+            connect(ptr, &QGuiApplication::paletteChanged, this, &Theme::systemPaletteChanged);
```

Comment:
It always returns a nullptr

Reply:
are you sure because I added a log and I get this
```
[ debug nextcloud.gui.systray /home/mgallien/work/nextcloud/desktop/src/gui/systray.cpp:74 ]    [ OCC::Systray::Systray ]:      OCC::Application(0x7fffb8a1af40) OCC::Application(0x7fffb8a1af40)
```

Reply:
It might be a valid QApplication pointer but in my testing when converting to a QGuiApplication pointer the result is always a nullptr

Reply:
My output:

```
QML debugging is enabled. Only use this in a safe environment.
THEME QObject(0x0)
```

Reply:
yeah we mindlessly trigger the creation of the `Theme` singleton to read a constant from it set at compilation time